src/stream/src/executor/top_n/top_n_cache.rs

```diff
@@ -108,7 +108,10 @@ impl<const WITH_TIES: bool> TopNCache<WITH_TIES> {
             low: BTreeMap::new(),
             middle: BTreeMap::new(),
             high: BTreeMap::new(),
-            high_capacity: (offset + limit) * TOPN_CACHE_HIGH_CAPACITY_FACTOR,
+            high_capacity: offset
+                .checked_add(limit)
+                .and_then(|v| v.checked_mul(TOPN_CACHE_HIGH_CAPACITY_FACTOR))
+                .unwrap_or(usize::MAX),
```

Comment:
So `offset` and `limit` are still `usize` here. Would this have the same overflow issue when `usize` is smaller than `i64`?

Reply:
It is indeed a problem. Because the `len()` of the caches need to be compared with `limit` and `offset`, the overflow problem in such scenarios seems unavoidable. 🤔

Reply:
Perhaps we should just return an error when overflow happens. After all, it feels ok to me to add some extra restrictions on mvs 🤣

Reply:
I have added a check which will panic when overflow occurs. We might figure out a more readable error message later.

Reply:
Do you mean running on a 32-bit machine? I think there's no need to worry about this and always assume on a 64-bit machine, or we're not able to cache all rows in the memory, then the whole implementation will be wrong. 🤔

Reply:
Ok, then a simple `as usize` will do.